File: Python/Test/jdi_uitests_webtests/test/composite/form_tests.py
from JDI.jdi_assert.testing.assertion import Assert
from Test.jdi_uitests_webtests.main.entities.contact import Contact
from Test.jdi_uitests_webtests.main.enums.entities import Buttons
from Test.jdi_uitests_webtests.main.enums.preconditions import Preconditions
from Test.jdi_uitests_webtests.main.page_objects.epam_jdi_site import EpamJDISite
from Test.jdi_uitests_webtests.main.utils.common_action_data import CommonActionsData
from Test.jdi_uitests_webtests.test.init_tests import InitTests


class FormTests(InitTests):
    form = EpamJDISite.contact_form_page.contact_form
    contact = Contact("Ivan", "Ivanov", "Smart Man")

    def setUp(self):
        Preconditions.CONTACT_PAGE.is_in_state()

    def test_fill(self):
        self.form.fill(self.contact)
        Assert.assert_equal(self.form.get_form_value(), {"Ivan", "Ivanov", "Smart Man"})

    def test_submit_form(self):
        self.form.submit_form(self.contact)
        Assert.assert_equal(EpamJDISite.contact_form_page.result.get_text(), str(self.contact))

    def test_submit_spec_button_string(self):
        self.form.submit_form(self.contact, button_name="submit")
        Assert.assert_equal(EpamJDISite.contact_form_page.result.get_text(), str(self.contact))

    def test_submit_speck_button_enum(self):
        self.form.submit_form(self.contact, Buttons.SUBMIT)
        Assert.assert_equal(EpamJDISite.contact_form_page.result.get_text(), str(self.contact))

    def test_submit_string(self):
        self.form.submit_form(self.contact.description)
        Assert.assert_equal(EpamJDISite.contact_form_page.result.get_text(), "Summary: 3\nDescription: {0}".format(self.contact.description))

    def test_verify(self):
        self.form.fill(self.contact)
        Assert.is_true(self.form.verify(self.contact))

    def test_check(self):
        self.form.fill(self.contact)
        self.form.check(self.contact)

File: Java/Tests/jdi-uitest-wintests/src/test/java/com/epam/jdi/uitests/testing/unittests/composite/FormTests.java
package com.epam.jdi.uitests.testing.unittests.composite;

import com.epam.jdi.uitests.testing.unittests.InitTests;
import com.epam.jdi.uitests.testing.unittests.enums.Buttons;
import com.epam.jdi.uitests.testing.unittests.pageobjects.forms.IContactForm;
import com.epam.web.matcher.testng.Assert;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import java.lang.reflect.Method;
import java.util.Arrays;
import java.util.List;
import java.util.function.Supplier;

import static com.epam.jdi.uitests.core.preconditions.PreconditionsState.isInState;
import static com.epam.jdi.uitests.testing.unittests.entities.Contact.DEFAULT;
import static com.epam.jdi.uitests.testing.unittests.enums.Preconditions.CONTACT_PAGE;
import static com.epam.jdi.uitests.testing.unittests.pageobjects.Desktop.mainWindow;
import static com.epam.web.matcher.testng.Assert.*;


public class FormTests extends InitTests {
    private Supplier<IContactForm> form = () -> mainWindow.mainTabPane.contactFormTab.nestedContactFormView.contactForm;

    public FormTests() {}

    public FormTests(Supplier<IContactForm> form) {
        this.form = form;
    }

    @BeforeMethod
    public void before(final Method method) {
        isInState(CONTACT_PAGE, method);
    }

    @Test
    public void fillTest() {
        form.get().fill(DEFAULT);
        assertEquals(form.get().getFormValue(), DEFAULT.toList());
    }

    @Test
    public void submitTest() {
        form.get().submit(DEFAULT);
        Assert.assertContains(Arrays.toString(mainWindow.passwordTextBox.getLines()), DEFAULT.toString());
    }

    @Test
    public void submitSpecButtonStringTest() {
        form.get().submit(DEFAULT, "submit");
        Assert.assertContains(Arrays.toString(mainWindow.passwordTextBox.getLines()), DEFAULT.toString());
    }

    @Test
    public void submitSpecButtonEnumTest() throws Exception {
        form.get().submit(DEFAULT, Buttons.SUBMIT);
        Assert.assertContains(Arrays.toString(mainWindow.passwordTextBox.getLines()), DEFAULT.toString());
    }

    @Test
    public void submitStringTest() throws Exception {
        form.get().submit(DEFAULT.name);
        Assert.assertContains(String.format("Summary: 3\nName: %s", DEFAULT.name), DEFAULT.name);
    }

    @Test
    public void verifyTest() {
        form.get().fillForm(DEFAULT);
        List<String> verify = form.get().verify(DEFAULT);
        isEmpty(verify);
    }

    @Test
    public void checkTest() {
        form.get().fillForm(DEFAULT);
        hasNoExceptions(() -> form.get().check(DEFAULT));
    }
}
